package/boot/uboot-sunxi/patches/253-sunxi-h5-add-support-for-nanopi-r1s.patch

```diff
@@ -0,0 +1,295 @@
+--- /dev/null
++++ b/arch/arm/dts/sun50i-h5-nanopi-r1s-h5.dts
+@@ -0,0 +1,265 @@
++// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
++/*
++ * Copyright (C) 2021 Chukun Pan <amadeus@jmu.edu.cn>
++ *
++ * Based on sun50i-h5-nanopi-neo-plus2.dts, which is:
++ *   Copyright (C) 2017 Antony Antony <antony@phenome.org>
++ *   Copyright (C) 2016 ARM Ltd.
++ */
++
++/dts-v1/;
++#include "sun50i-h5.dtsi"
++
++#include <dt-bindings/gpio/gpio.h>
++#include <dt-bindings/input/input.h>
++
++/ {
++	model = "FriendlyARM NanoPi R1S H5";
++	compatible = "friendlyarm,nanopi-r1s-h5", "allwinner,sun50i-h5";
++
++	aliases {
++		ethernet0 = &emac;
++		ethernet1 = &rtl8189etv;
++		serial0 = &uart0;
++	};
++
++	chosen {
++		stdout-path = "serial0:115200n8";
++	};
++
++	leds {
++		compatible = "gpio-leds";
++
++		led-0 {
++			label = "nanopi:green:lan";
```

Comment:
label is deprecated upstream. Please switch to color and function.

Reply:
No, please just drop the model prefix and keep the label. We do not support color/function yet in OpenWrt.

https://github.com/openwrt/openwrt/pull/3437

Reply:
fixed

Reply:
Err, I just realized that this is for uboot. So, probably nothing speaks against using color/function here, if you know how to use it properly.

In contrast, the relevant discussion would be what to do with the target patch. Is that one already upstreamed?
Yes -> Put the kernel version in the file name.
No -> One should consider to use label here. However, as long as no aliases are used, this will probably just work as expected with color/function.